### DISCUSSION TITLE
PWM different frequencies

### DISCUSSION THREAD

**Discussion by nmrds** [arendst/Tasmota#17986] [Q&A]
Hello,
I need to control two devices that require different PWM frequencies, and they are connected to one ESP32.
One frequency needs to be 500Hz and the other 4kHz.
I'm using Scripting Language, and the commands pwmN(-pin freq) and pwmN(val).

>\>B
; define pwm for pins 13 to 500 Hz and 34 to 4kHz
pwm1(-13 500)
pwm2(-34 4000)

if I use them independently each device works well, but when I use both, the frequency of pwm1 seems to also be set to 4kHz.
If the command pwmN(-pin freq) has the freq parameter, for me it means that I can set a different frequency for each pwm.
Is there a way to implement this in Scripting or am I doing something wrong?

Thanks

**Comment by barbudor** [comment]
It's not about scripting, it's about PWM capabilities
On esp8266 at least, 1 frequency to rule them all
You can't achieve that

**Reply by nmrds** [reply]
I'm using one ESP32. Sorry, forgot to mention that.
